dataservice/api/genomic_file/resources.py

```diff
@@ -147,6 +155,9 @@ def delete(self, kf_id):
         if gf is None:
             abort(404, 'could not find {} `{}`'.format('GenomicFile', kf_id))
 
+        # Delete genomic file from indexd
+        indexd.delete_record(gf)
```

Comment:
I'd like to see these operations masked behind the GenomicFile object so that we maintain the same interface among all of our entities. It also helps in the case that we ever want to work with GenomicFiles in another setting where we will likely expect to do the same operations and would likely need to duplicate this logic there as well.

Reply:
Hmm well I think the ORM layer shouldn’t really be making http calls. We’re probably going to have other instances where we’re integrating with 3rd party APIs to (i.e. Cavatica to get workflow info). I would think that we would want to encapsulate the code for interfacing with a 3rd party API in its own module. And keep it out of the the ORM class  even if there is some duplicate code

Reply:
This isn't really in the ORM layer. The indexd extension sort of creates it's own ORM which maps a subset of the object's fields to indexd and hooks into SQLalchemy's events to syncronize changes. If we want more functionality out of the indexd, it may be better to abstract it out into a client, but seeing as it's rather limited, it seems like it's not yet appropriate.